lib/oli/automation_setup.ex

```diff
@@ -0,0 +1,354 @@
+defmodule Oli.AutomationSetup do
+  @moduledoc """
+  Sets up data in a torus instance that is meant be used with automated e2e tests to
+  enable testing of content in remote torus instances.
+  """
+  alias Lti_1p3.Tool.ContextRoles
+
+  alias Oli.Repo
+  alias Oli.Authoring.Course.Project
+  alias Oli.Authoring.Course.ProjectResource
+
+  import Ecto.Query, warn: false
+  require Logger
+
+  def setup_data(
+        project_archive,
+        create_learner?,
+        create_educator?,
+        create_author?,
+        create_section?
+      ) do
+    with {:ok, author, author_password} <-
```

Comment:
Maybe we can try to make this a transaction? If some of the steps fail, we can rollback changes. Ecto.Multi sounds like a good friend here

Reply:
Got that, and also did project teardown in a transaction.